airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/stream_reader.py

```diff
@@ -79,17 +86,92 @@ def get_matching_files(
                     )
 
     def open_file(self, file: RemoteFile, mode: FileReadMode, encoding: Optional[str], logger: logging.Logger) -> IOBase:
-        if not self.use_file_transfer():
-            remote_file = self.sftp_client.sftp_connection.open(file.uri, mode=mode.value)
-        else:
-            remote_file = self.sftp_client.sftp_connection.open(file.uri, mode=mode.value, bufsize=262144)
-            # prefetch() works by requesting multiple blocks of data in advance,
-            # rather than waiting for one block to be retrieved before requesting the next.
-            # This is a boost for reading but can cause memory errors, should be removed
-            # when we work on https://github.com/airbytehq/airbyte-internal-issues/issues/10480
-            remote_file.prefetch(remote_file.stat().st_size)
+        remote_file = self.sftp_client.sftp_connection.open(file.uri, mode=mode.value)
         return remote_file
 
+    @staticmethod
+    def create_progress_handler(local_file_path: str, logger: logging.Logger):
+        previous_bytes_copied = 0
+
+        def progress_handler(bytes_copied, total_bytes):
+            nonlocal previous_bytes_copied
+            if bytes_copied - previous_bytes_copied >= 100 * 1024 * 1024:
+                logger.info(
+                    f"{bytes_copied / (1024 * 1024):,.2f} MB ({bytes_copied / (1024 * 1024 * 1024):.2f} GB) "
+                    f"of {total_bytes / (1024 * 1024):,.2f} MB ({total_bytes / (1024 * 1024 * 1024):.2f} GB) "
+                    f"written to {local_file_path}"
+                )
+                previous_bytes_copied = bytes_copied
+
+                # Get available disk space
+                disk_usage = psutil.disk_usage("/")
+                available_disk_space = disk_usage.free
+
+                # Get available memory
+                memory_info = psutil.virtual_memory()
+                available_memory = memory_info.available
+                logger.info(
+                    f"Available disk space: {available_disk_space / (1024 * 1024):,.2f} MB ({available_disk_space / (1024 * 1024 * 1024):.2f} GB), "
+                    f"available memory: {available_memory / (1024 * 1024):,.2f} MB ({available_memory / (1024 * 1024 * 1024):.2f} GB)."
+                )
+
+        return progress_handler
+
+    @override
+    def get_file(self, file: RemoteFile, local_directory: str, logger: logging.Logger) -> Dict[str, str | int]:
```

Comment:
[5GB >](https://cloud.airbyte.com/workspaces/166f3e04-9021-4f63-a963-a43abb548cec/connections/7e483cf1-3b9a-4e5b-8c4f-c0b74622f5b3/timeline?openLogs=true&eventId=6b7961d3-68d6-41e9-a971-cf39e3362f85) transfer looks good in the testing workspace.

Reply:
Could you please add local integration test to ensure memory usage does not exceed limits? 

P.S. see `pytest.mark.limit_memory`

Reply:
@artem1205 tests are now there. Thanks for pointing this!